### PR TITLE
cut/paste AccessorProperty and PandasDelegate to core.accessor

### PR DESCRIPTION
**Pull request by jbrockmendel**
To try to get some positive momentum on wrapping up #17042, this PR cut/pastes AccessorProperty and PandasDelegate from core.base to core.accessor, updates the appropriate imports.

The idea is that it will be easier to review actual design changes in isolation of cut/paste changes.

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
looks fine; ping when green

**Comment by jbrockmendel** [comment]
Just pushed a fix for the current issue.  Keep the implementation of PandasDelegate in core.accessor, but define the final class in core.base to avoid a circular import.

File: pandas/core/base.py [hunk 1]
```diff
@@ -153,98 +154,8 @@ def __setattr__(self, key, value):
...
-            # this ensures that Series.str.<method> is well defined
-            return self.accessor_cls
-        return self.construct_accessor(instance)
-
-    def __set__(self, instance, value):
-        raise AttributeError("can't set attribute")
-
-    def __delete__(self, instance):
-        raise AttributeError("can't delete attribute")
+class PandasDelegate(_PandasDelegate, PandasObject):
```
Comment (jreback) [inline pandas/core/base.py]: not a fan of this
rather u can simply directly import core.accessor and use it as a mix in class where needed

**Comment by jreback** [comment]
thanks!

**Pull request merged**